config/opal_check_package2.m4

```diff
@@ -0,0 +1,237 @@
+           $6],
+          [$7])
+
+    AS_VAR_POPDEF([opal_Lib])dnl
+])
+
+
+dnl OPAL_CHECK_PACKAGE(prefix,
+dnl                    header,
+dnl                    library,
+dnl                    function,
+dnl                    extra-libraries,
+dnl                    package,
+dnl                    [action-if-found], [action-if-not-found],
+dnl                    includes)
+dnl -----------------------------------------------------------
+dnl Check for package defined by header and libs, and probably
+dnl located in dir-prefix, possibly with libs in libdir-prefix.
+dnl Both dir-prefix and libdir-prefix can be empty.  Will set
+dnl prefix_{CPPFLAGS, LDFLAGS, LIBS} as needed.
+dnl
+dnl The general intent of this macro is to provide finer-grained scoping
+dnl of C preprocessor flags, linker flags, and libraries (as opposed to
+dnl unconditionally adding to the top-level CPFLAGS, LDFLAGS, and LIBS,
+dnl which get used to compile/link *everything*).
+dnl
+dnl Here is a breakdown of the parameters:
+dnl
+dnl * prefix: the macro sets $prefix_CPPFLAGS, $prefix_LDFLAGS, and
+dnl   $prefix_LIBS (and AC_SUBSTs all of them).  For example, if a
+dnl   provider uses this macro to check for a header/library that it
+dnl   needs, it might well set prefix to be its provider name.
+dnl * header_filename: the foo.h file to check for
+dnl * library_name / function_name: check for function function_name in
+dnl   -llibrary_name.  Specifically, for library_name, use the "foo" form,
+dnl   as opposed to "libfoo".
+dnl * extra_libraries: if the library_name you are checking for requires
+dnl   additonal -l arguments to link successfully, list them here.
+dnl * dir_prefix: if the header/library is located in a non-standard
+dnl   location (e.g., /opt/foo as opposed to /usr), list it here
+dnl * libdir_prefix: if the library is not under $dir_prefix/lib or
+dnl   $dir_prefix/lib64, list it here.
+dnl * action_if_found: if both the header and library are found and
+dnl   usable, execute action_if_found
+dnl * action_if_not_found: otherwise, execute action_if_not_found
+dnl * extra_includes: if including header_filename requires additional
+dnl   headers to be included first, list them here
+dnl
+dnl The output _CPPFLAGS, _LDFLAGS, and _LIBS can be used to limit the
+dnl scope various flags in Makefiles.
+dnl
+AC_DEFUN([OPAL_CHECK_PACKAGE2],[
+    OPAL_VAR_SCOPE_PUSH([opal_check_package2_happy opal_check_package2_cppflags opal_check_package_$1_save_CPPFLAGS opal_check_package_$1_save_LDFLAGS opal_check_package_$1_save_LIBS opal_check_package_$1_orig_CPPFLAGS opal_check_package_$1_orig_LDFLAGS opal_check_package_$1_orig_LIBS])
+    opal_check_package_$1_save_CPPFLAGS="$CPPFLAGS"
+    opal_check_package_$1_save_LDFLAGS="$LDFLAGS"
+    opal_check_package_$1_save_LIBS="$LIBS"
+
+    opal_check_package_$1_orig_CPPFLAGS="$$1_CPPFLAGS"
+    opal_check_package_$1_orig_LDFLAGS="$$1_LDFLAGS"
+    opal_check_package_$1_orig_LIBS="$$1_LIBS"
+
+    AS_IF([test "$with_$6" != "no"],
```

Comment:
This code would be a lot simpler if you defined a variable foobar=$with_$6 and used that in all the tests instead of constantly doing the m4 expansion thing.